File: kickkart-backend/src/main/java/com/kickkart/service/ProductService.java
package com.kickkart.service;

import com.kickkart.dto.request.ProductRequest;
import com.kickkart.dto.response.ProductResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ProductService {

    ProductResponse createProduct(ProductRequest request);

    Page<ProductResponse> getAllProducts(
            String search,
            String category,
            int page,
            int size,
            String sort,
            String direction
    );
    ProductResponse getProductById(Long id);

    ProductResponse updateProduct(Long id, ProductRequest request);

    void deleteProduct(Long id);
}

File: kickkart-frontend/src/api/productService.js
import api from "./axios";

export const getProducts = async ({
    page = 0,
    size = 12,
    search = "",
    category = "",
    sort = "id",
    direction = "desc",
}) => {

    const response = await api.get("/products", {
        params: {
            page,
            size,
            search,
            category,
            sort,
            direction,
        },
    });

    return response.data;
};

export const getProductById = async (id) => {

    const response = await api.get(`/products/${id}`);

    return response.data;

};